### ISSUE TITLE
request or response defines object inline instead of in separate schema ref - produces None class

### ISSUE THREAD

**Issue by martin0** [asyncee/swagger_codegen#15]
the included json includes a few samples of this issue.
`{
    "openapi": "3.0.0",
    "info": {
        "title": "Client Portal Web API",
        "description": "Production version of the Client Portal Web API",
        "version": "1.0.0"
    },
    "paths": {
        "/iserver/accounts": {
            "get": {
                "summary": "Brokerage Accounts",
                "description": "Returns a list of accounts the user has trading access to, their respective aliases and the currently selected account. Note this endpoint must be called before modifying an order or querying open orders.",
                "tags": [
                    "Account"
                ],
                "responses": {
                    "200": {
                        "description": "An array of accounts",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "type": "object",
                                    "properties": {
                                        "accounts": {
                                            "type": "array",
                                            "items": {
                                                "type": "string"
                                            },
                                            "description": "Unique account id"
                                        },
                                        "aliases": {
                                            "type": "object",
                                            "description": "Account Id and its alias"
                                        },
                                        "selectedAccount": {
                                            "type": "string"
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        },
        "/iserver/marketdata/snapshot": {
            "get": {
                "summary": "Market Data",
                "description": "Get Market Data for the given conid(s). The end-point will return by default bid, ask, last, change, change pct, close, listing exchange.\nSee response fields for a list of available fields that can be request via fields argument.\nThe endpoint /iserver/accounts should be called prior to /iserver/marketdata/snapshot.\nTo receive all available fields the /snapshot endpoint will need to be called several times.\n",
                "tags": [
                    "Market_Data"
                ],
                "parameters": [
                    {
                        "in": "query",
                        "name": "conids",
                        "description": "list of conids separated by comma",
                        "required": true,
                        "schema": {
                            "type": "string"
                        }
                    },
                    {
                        "in": "query",
                        "name": "since",
                        "description": "time period since which updates are required. uses epoch time with milliseconds.",
                        "required": false,
                        "schema": {
                            "type": "integer"
                        }
                    },
                    {
                        "in": "query",
                        "name": "fields",
                        "description": "list of fields separated by comma",
                        "required": false,
                        "schema": {
                            "type": "string"
                        }
                    }
                ],
                "responses": {
                    "200": {
                        "description": "Returns an array of objects",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "type": "array",
                                    "items": {
                                        "type": "object",
                                        "properties": {
                                            "31": {
                                                "type": "string",
                                                "description": "Last Price"
                                            },
                                            "55": {
                                                "type": "string",
                                                "description": "Symbol"
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    },
                    "400": {
                        "description": "sent when accounts are not queried before sending this request",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "type": "object",
                                    "properties": {
                                        "error": {
                                            "type": "string"
                                        },
                                        "statusCode": {
                                            "type": "integer"
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        },
        "/iserver/secdef/search": {
            "post": {
                "summary": "Search by symbol or name",
                "tags": [
                    "Contract"
                ],
                "requestBody": {
                    "content": {
                        "application/json": {
                            "schema": {
                                "type": "object",
                                "required": [
                                    "symbol"
                                ],
                                "properties": {
                                    "symbol": {
                                        "type": "string",
                                        "description": "symbol or name to be searched"
                                    },
                                    "name": {
                                        "type": "boolean",
                                        "description": "should be true if the search is to be performed by name. false by default."
                                    },
                                    "secType": {
                                        "type": "string",
                                        "description": "If search is done by name, only the assets provided in this field will be returned. Currently, only STK is supported."
                                    }
                                }
                            }
                        }
                    },
                    "description": "symbol or name to be searched",
                    "required": true
                },
                "responses": {
                    "200": {
                        "description": "returns an array of results",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "type": "array",
                                    "items": {
                                        "type": "object",
                                        "properties": {
                                            "conid": {
                                                "type": "integer"
                                            },
                                            "sections": {
                                                "type": "array",
                                                "items": {
                                                    "type": "object"
                                                }
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    },
                    "500": {
                        "description": "error while processing the request",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "type": "object",
                                    "properties": {
                                        "error": {
                                            "type": "string"
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        },
        "/iserver/scanner/params": {
            "get": {
                "summary": "get lists of available scanners",
                "description": "Returns an object contains four lists contain all parameters for scanners",
                "tags": [
                    "Scanner"
                ],
                "responses": {
                    "200": {
                        "description": "An object contains lists",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "type": "object",
                                    "properties": {
                                        "scan_type_list": {
                                            "type": "array",
                                            "items": {
                                                "type": "object",
                                                "properties": {
                                                    "display_name": {
                                                        "type": "string"
                                                    },
                                                    "code": {
                                                }
                                            }
                                        },
                                        "instrument_list": {
                                            "type": "array",
                                            "items": {
                                                "type": "object",
                                                "properties": {
                                                    "display_name": {
                                                        "type": "string"
                                                    },
                                                    "type": {
                                                        "type": "string"
                                                    },
                                                    "filters": {
                                                        "type": "array",
                                                        "items": {
                                                            "type": "string",
                                                            "description": "code of filter"
                                                        }
                                                    }
                                                }
                                            }
                                        },
                                        "filter_list": {
                                            "type": "array",
                                            "items": {
                                                "type": "object",
                                                "properties": {
                                                    "group": {
                                                        "type": "string"
                                                    }
                                                }
                                            }
                                        },
                                        "location_tree": {
                                            "type": "array",
                                            "items": {
                                                "type": "object",
                                                "properties": {
                                                    "display_name": {
                                                        "type": "string"
                                                    },
                                                    "type": {
                                                        "type": "string"
                                                    },
                                                    "locations": {
                                                        "type": "array",
                                                        "items": {
                                                            "type": "object",
                                                            "properties": {
                                                                "display_name": {
                                                                    "type": "string"
                                                                },
                                                                "type": {
                                                                    "type": "string"
                                                                }
                                                            }
                                                        }
                                                    }
                                                }
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        },
    },
    "servers": [
        {
            "url": "https://localhost:5000/v1/portal"
        }
    ],
    "components": {
        "requestBodies": {
            "Body": {
                "content": {
                    "application/json": {
                        "schema": {
                            "type": "object",
                            "properties": {
                                "acctIds": {
                                    "type": "array",
                                    "items": {
                                        "type": "string",
                                        "description": "account id"
                                    }
                                }
                            }
                        }
                    }
                },
                "description": "an array of account ids",
                "required": true
            }
        },
        "schemas": {
        }
    }
}`

**Comment by martin0** [comment]
Any chance of fixing this? Should be similar to the issue fixed most recently. Much appreciated

**Comment by asyncee** [comment]
I'm quite busy right now, sorry. I hope that i will fix this on the week.

**Comment by martin0** [comment]
thanks for setting my expectations. Your time is much appreciated :-)

**Comment by martin0** [comment]
there's some issues with the json i provided. I'll correct and resubmit...

**Comment by martin0** [comment]
`{
    "openapi": "3.0.0",
    "info": {
        "title": "Client Portal Web API",
        "description": "Production version of the Client Portal Web API",
        "version": "1.0.0"
    },
    "paths": {
        "/iserver/account/orders": {
            "get": {
                "summary": "Live Orders",
                "description": "The end-point is meant to be used in polling mode, e.g. requesting every x seconds.\nThe response will contain two objects, one is notification, the other is orders. \nOrders is the list of orders (cancelled, filled, submitted) with activity in the current day. \nNotifications contains information about execute orders as they happen, see status field.\n",
                "tags": [
                    "Order"
                ],
                "responses": {
                    "200": {
                        "description": "An object contains two arrays",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "type": "object",
                                    "properties": {
                                        "notifications": {
                                            "type": "array",
                                            "items": {
                                                "type": "object",
                                                "description": "contains notifications for different status of orders, the data structure is same to orders"
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        },
        "/iserver/accounts": {
            "get": {
                "summary": "Brokerage Accounts",
                "description": "Returns a list of accounts the user has trading access to, their respective aliases and the currently selected account. Note this endpoint must be called before modifying an order or querying open orders.",
                "tags": [
                    "Account"
                ],
                "responses": {
                    "200": {
                        "description": "An array of accounts",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "type": "object",
                                    "properties": {
                                        "accounts": {
                                            "type": "array",
                                            "items": {
                                                "type": "string"
                                            },
                                            "description": "Unique account id"
                                        },
                                        "aliases": {
                                            "type": "object",
                                            "description": "Account Id and its alias"
                                        },
                                        "selectedAccount": {
                                            "type": "string"
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    },
    "servers": [
        {
            "url": "https://localhost:5000/v1/portal"
        }
    ],
    "components": {
        "requestBodies": {
            "Body": {
                "content": {
                    "application/json": {
                        "schema": {
                            "type": "object",
                            "properties": {
                                "acctIds": {
                                    "type": "array",
                                    "items": {
                                        "type": "string",
                                        "description": "account id"
                                    }
                                }
                            }
                        }
                    }
                },
                "description": "an array of account ids",
                "required": true
            }
        },
        "schemas": {
            "authStatus": {
                "type": "object",
                "properties": {
                    "authenticated": {
                        "description": "Brokerage session is authenticated",
                        "type": "boolean"
                    }
                }
            }
    
        }
    }
}`

**Comment by asyncee** [comment]
Can you reformat json, it's still looks unformatted. You can use triple backticks (```) to create a block of code.

**Comment by martin0** [comment]
```{
    "openapi": "3.0.0",
    "info": {
        "title": "Client Portal Web API",
        "description": "Production version of the Client Portal Web API",
        "version": "1.0.0"
    },
    "paths": {
        "/iserver/account/orders": {
            "get": {
                "summary": "Live Orders",
                "description": "The end-point is meant to be used in polling mode, e.g. requesting every x seconds.\nThe response will contain two objects, one is notification, the other is orders. \nOrders is the list of orders (cancelled, filled, submitted) with activity in the current day. \nNotifications contains information about execute orders as they happen, see status field.\n",
                "tags": [
                    "Order"
                ],
                "responses": {
                    "200": {
                        "description": "An object contains two arrays",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "type": "object",
                                    "properties": {
                                        "notifications": {
                                            "type": "array",
                                            "items": {
                                                "type": "object",
                                                "description": "contains notifications for different status of orders, the data structure is same to orders"
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        },
        "/iserver/accounts": {
            "get": {
                "summary": "Brokerage Accounts",
                "description": "Returns a list of accounts the user has trading access to, their respective aliases and the currently selected account. Note this endpoint must be called before modifying an order or querying open orders.",
                "tags": [
                    "Account"
                ],
                "responses": {
                    "200": {
                        "description": "An array of accounts",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "type": "object",
                                    "properties": {
                                        "accounts": {
                                            "type": "array",
                                            "items": {
                                                "type": "string"
                                            },
                                            "description": "Unique account id"
                                        },
                                        "aliases": {
                                            "type": "object",
                                            "description": "Account Id and its alias"
                                        },
                                        "selectedAccount": {
                                            "type": "string"
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    },
    "servers": [
        {
            "url": "https://localhost:5000/v1/portal"
        }
    ],
    "components": {
        "requestBodies": {
            "Body": {
                "content": {
                    "application/json": {
                        "schema": {
                            "type": "object",
                            "properties": {
                                "acctIds": {
                                    "type": "array",
                                    "items": {
                                        "type": "string",
                                        "description": "account id"
                                    }
                                }
                            }
                        }
                    }
                },
                "description": "an array of account ids",
                "required": true
            }
        },
        "schemas": {
            "authStatus": {
                "type": "object",
                "properties": {
                    "authenticated": {
                        "description": "Brokerage session is authenticated",
                        "type": "boolean"
                    }
                }
            }
    
        }
    }
}```

**Comment by martin0** [comment]
nb, although it doesn't reference the authStatus schema (at the end), i had to leave something in, otherwise it complained lwhen loading the file

**Comment by asyncee** [comment]
Fixed in 0.1.27, thanks.

**Comment by martin0** [comment]
Great -thanks - I'll give it a go :-)
